[WD-7737] desktop features rebrand

akbarkz opened the pull request:
## Done

- Rebranded /desktop/features according to [Figma design](https://www.figma.com/file/wLOtPeQhAiExhQCwKYhi3H/24.04-U.com---desktop-(rebranding)?type=design&node-id=888-10097&mode=design&t=g2XHjM3FGL8B1bSo-0)

## QA

- Navigate to /desktop/features and check the page
   - Be sure to test on mobile, tablet and desktop screen sizes

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-7737

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)

webteam-app commented:
Demo starting at https://ubuntu-com-13483.demos.haus

juanruitina commented:
Looks good. Some comments:

* I just noticed the Firefox description was wrong on the Figma file and outdated on the live page. Please change it to "A powerful, extensible web browser with support for modern web application technologies.", I updated it both on Figma and [copy doc](https://docs.google.com/document/d/1nLHJlZ3igDwAAOUDP03wp4iKLxZts8110EA2ZOF5uaU/edit).
* The Spotify, Firefox and Chrome icons are outdated, please replace with the ones MariaPaula uploaded to the [assets manager](https://assets.ubuntu.com/manager?tag=desktop-icons).
* The photos section (where the community photos are) looks a bit broken on small and medium viewports. Photos are not visible but credits remain. I don't have strong opinions on whether to keep this section or hide it on smaller viewports, but if we hide it, let's hide it the whole section at once. (@lyubomir-popov please confirm if you think we should hide it or not)
* The list of apps needs a bit of work on small viewports. At a minimum I'd hide the HRs we have every three of them and add some spacing (see below), but I defer to @lyubomir-popov on the final approach.

![ubuntu-com-13483 demos haus_desktop_features](https://github.com/canonical/ubuntu.com/assets/1409729/0e047e2d-0106-4e9f-8b39-42d73ac745b3)

akbarkz commented:
Thansk @juanruitina for the review! I addressed all of your remarks.
With the Photos section I hid the photo and the credits text, but left the title and description on small screen.
With the Apps icons I changed the layout on medium and small screens for it to look better. @lyubomir-popov please let me know if I still need to change it.

juanruitina commented:
It looks great now, UX+1!

juanruitina commented:
Fine by me, UX of this is OK.

lyubomir-popov commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,593 @@
-      <h2>A whole world of apps</h2>
-      <p >Ubuntu offers thousands of apps available for download. Most are available for free and can be installed with just a few clicks.</p>
+<section class="p-strip is-shallow">
+  <div class="p-section">
```
```suggestion
  <div class="p-section--shallow">
```

lyubomir-popov commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,593 @@
-          <p class="p-matrix__desc">A fast and secure desktop messaging app, perfectly synced with your mobile phone.</p>
+  <div>
+    <div class="row--25-75">
+      <div class="col"></div>
```
this is unnecesary - when there's only one, it goes to the right

lyubomir-popov reviewed (approved):
LGTM, with one suggestion. I would replace the shallow p-strips with p-sections. Not a big issue, but the correct way is p-sections, theuy add 4rem padding bottom, while two adjacent p-strip is -shallow divs add up to 3rem (1.5rem *2).

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
+            <div class="row">
+              <div class="col-6 col-medium-6 col-small-1">
+                <div class="p-image-wrapper">
+                  {{
+                  image(
+                  url="https://assets.ubuntu.com/v1/9917aba6-spotify-linux-256.png",
+                  alt="Spotify icon",
+                  height="32",
+                  width="32",
+                  hi_def=True,
+                  ) | safe
+                  }}
```
```suggestion
                  {{
                    image(
                    url="https://assets.ubuntu.com/v1/9917aba6-spotify-linux-256.png",
                    alt="Spotify icon",
                    height="32",
                    width="32",
                    hi_def=True,
                    ) | safe
                  }}
```

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
+            <div class="row">
+              <div class="col-6 col-medium-6  col-small-1">
+                <div class="p-image-wrapper">
+                  {{
+                  image(
+                  url="https://assets.ubuntu.com/v1/4bb576dc-bitwarden.svg",
+                  alt="Bitwarden icon",
+                  height="32",
+                  width="32",
+                  hi_def=True,
+                  attrs={"class": "u-vertically-center"},
+                  ) | safe
+                  }}
```
```suggestion
                  {{
                    image(
                    url="https://assets.ubuntu.com/v1/4bb576dc-bitwarden.svg",
                    alt="Bitwarden icon",
                    height="32",
                    width="32",
                    hi_def=True,
                    attrs={"class": "u-vertically-center"},
                    ) | safe
                  }}
```

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
+            <div class="row">
+              <div class="col-8 col-medium-6 col-small-1">
+                <div class="p-image-wrapper">
+                  {{
+                  image(
+                  url="https://assets.ubuntu.com/v1/7b460167-vlc-icon.png",
+                  alt="VLC icon",
+                  height="32",
+                  width="32",
+                  hi_def=True,
+                  attrs={"class": "u-vertically-center"},
+                  ) | safe
+                  }}
```
```suggestion
                  {{
                    image(
                    url="https://assets.ubuntu.com/v1/7b460167-vlc-icon.png",
                    alt="VLC icon",
                    height="32",
                    width="32",
                    hi_def=True,
                    attrs={"class": "u-vertically-center"},
                    ) | safe
                  }}
```

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
+            <div class="row">
+              <div class="col-8 col-medium-6 col-small-1">
+                <div class="p-image-wrapper">
+                  {{
+                  image(
+                  url="https://assets.ubuntu.com/v1/8daf0eb5-firefox_logo.png",
+                  alt="Firefox icon",
+                  height="32",
+                  width="32",
+                  hi_def=True,
+                  attrs={"class": "u-vertically-center"},
+                  ) | safe
+                  }}
```
```suggestion
                  {{
                    image(
                    url="https://assets.ubuntu.com/v1/8daf0eb5-firefox_logo.png",
                    alt="Firefox icon",
                    height="32",
                    width="32",
                    hi_def=True,
                    attrs={"class": "u-vertically-center"},
                    ) | safe
                  }}
```

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
+            <div class="row">
+              <div class="col-8 col-medium-6 col-small-1">
+                <div class="p-image-wrapper">
+                  {{
+                  image(
+                  url="https://assets.ubuntu.com/v1/8a2851df-slack-icon-vanilla.svg",
+                  alt="Slack icon",
+                  height="32",
+                  width="32",
+                  hi_def=True,
+                  attrs={"class": "u-vertically-center"},
+                  ) | safe
+                  }}
```
```suggestion
                  {{
                    image(
                    url="https://assets.ubuntu.com/v1/8a2851df-slack-icon-vanilla.svg",
                    alt="Slack icon",
                    height="32",
                    width="32",
                    hi_def=True,
                    attrs={"class": "u-vertically-center"},
                    ) | safe
                  }}
```

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
+            <div class="row">
+              <div class="col-8 col-medium-6 col-small-1">
+                <div class="p-image-wrapper">
+                  {{
+                  image(
+                  url="https://assets.ubuntu.com/v1/5aeafc59-Chromium_Logo.svg",
+                  alt="Chromium icon",
+                  height="32",
+                  width="32",
+                  hi_def=True,
+                  attrs={"class": "u-vertically-center"},
+                  ) | safe
+                  }}
```
```suggestion
                  {{
                    image(
                    url="https://assets.ubuntu.com/v1/5aeafc59-Chromium_Logo.svg",
                    alt="Chromium icon",
                    height="32",
                    width="32",
                    hi_def=True,
                    attrs={"class": "u-vertically-center"},
                    ) | safe
                  }}
```

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
-                hi_def=True,
-                loading="lazy",
-                attrs={"class": "p-logo-section__logo"},
-              ) | safe
+            image(
+            url="https://assets.ubuntu.com/v1/4d47a674-gimp-60x43.svg",
+            alt="Gimp icon",
+            width="80",
+            hi_def=True,
+            loading="lazy",
+            ) | safe
             }}
```
Indentation

akbarkz replied:
This has been done automatically by the formatter

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
-                hi_def=True,
-                attrs={"class": "p-logo-section__logo"},
-                loading="lazy"
-              ) | safe
+            image(
+            url="https://assets.ubuntu.com/v1/786ed546-blender.svg",
+            alt="Blender icon",
+            width="80",
+            hi_def=True,
+            loading="lazy"
+            ) | safe
             }}
```
Indentation

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
-        height="768",
-        hi_def=True,
-        loading="lazy"
-        ) | safe
+      url="https://assets.ubuntu.com/v1/de158d4e-desktop-22.04-video.jpg",
+      alt="",
+      width="1360",
+      height="768",
+      hi_def=True,
+      loading="lazy"
+      ) | safe
       }}
```
Indentation

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
+  </div>
+  <div class="row--25-75">
+    <div class="col">
+      {{ image (
+      url="https://assets.ubuntu.com/v1/1d582338-dota2.jpg",
+      alt="",
+      width="1360",
+      height="768",
+      hi_def=True,
+      loading="lazy"
+      ) | safe
+      }}
```
Indentation

mtruj013 commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
+                </div>
+              </div>
+              <div class="col-8 col-medium-6 col-small-3">
+                <h3 class="p-heading--4 u-sv-2"><a href="https://snapcraft.io/vlc">VLC player</a></h3>
```
```suggestion
                <h3 class="p-heading--4 u-sv-2"><a href="https://snapcraft.io/vlc">VLC</a></h3>
```
Just VLC on the doc

akbarkz replied:
Iirc I was told by @juanruitina that the text and most of the images (if not specified otherwise in Figma design) should stay the same. And this one is an old text from how it was before. Juan, please let me know if I need to change this.

mtruj013 reviewed (approved):
Thanks @akbarkz! Some suggestions / comments in the code. I also left a couple of comments on the doc for @juanruitina regarding minor copy discrepancies in the footer, but shouldn't be a blocker

akbarkz commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
 {% block title %}Desktop features{% endblock %}
-{% block meta_description %}Learn about all the great features and default applications in the Ubuntu desktop operating system.{% endblock meta_description %}
-{% block meta_copydoc %}https://docs.google.com/document/d/1nLHJlZ3igDwAAOUDP03wp4iKLxZts8110EA2ZOF5uaU/edit{% endblock meta_copydoc %}
+{% block meta_description %}Learn about all the great features and default applications in the Ubuntu desktop operating
+system.{% endblock %}
```
It's a formatter to make the code not exceed 120 symbols horizontally. It has been done automatically to make the code more readable by eliminating the need to scroll horizontally for long pieces of text. We will discuss this on our dev stuff meeting next week.

akbarkz commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
-        loading="auto"
-        ) | safe
-      }}
+<style>
+  .image-inside-another-image {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    width: 70%;
+  }
+</style>
```
why so? this style is specific to this page only.

akbarkz commented on templates/desktop/features.html:
```diff
@@ -1,384 +1,586 @@
+  </div>
+  <div class="row--25-75">
+    <div class="col" style="position: relative">
+      {{
```
Next time you can just add one comment and say "fix it everywhere else as well" :)
